### PR TITLE
web: switch HUD tests to use react test-utils instead of enzyme

### PR DESCRIPTION
**Pull request by nicks**
Hello @lizzthabet,

Please review the following commits I made in branch nicks/reacttesting:

a729e8594ffe49a65515beb5ffa2e323e97c2624 (2022-04-19 14:31:46 -0400)
web: switch HUD tests to use react test-utils instead of enzyme

Code review reminders, by giving a LGTM you attest that:

* Commits are adequately tested
* Code is easy to understand and conforms to style guides
* Incomplete code is marked with TODOs
* Code is suitably instrumented with logging and metrics

### CODE REVIEW

**Comment by nicks** [comment]
react-testing-library doesn't have a good way to get the component instance, but this lower-level library does, so we'll just use that for tests that need it.

**Review by lizzthabet — APPROVED** [review]
🎉

File: web/src/HUD.test.tsx [hunk 1]
```diff
@@ -40,44 +43,36 @@ const emptyHUD = () => {
...
-  ReactDOM.render(emptyHUD(), div)
-  ReactDOM.unmountComponentAtNode(div)
-})
-
 it("renders reconnecting bar", async () => {
-  const root = mount(emptyHUD())
-  const hud = root.find(HUD)
-  expect(hud.text()).toEqual(expect.stringContaining("Loading"))
+  let root = emptyHUD()
+  let rootTree = renderIntoDocument(root)
```
Comment (lizzthabet) [inline web/src/HUD.test.tsx:53]: oh! it looks like this may not return the tree. i'm seeing type errors locally and in ci.

Comment (lizzthabet) [inline web/src/HUD.test.tsx:53]: oh! it looks like this may not return the tree. i'm seeing type errors locally and in ci.

Comment (nicks) [reply web/src/HUD.test.tsx]: fixed! tbh i don't know why react thinks this returns void

**Pull request merged**